Source/WebGPU/WebGPU/RenderPassEncoder.mm

```diff
@@ -722,15 +731,21 @@ static void setViewportMinMaxDepthIntoBuffer(auto& fragmentDynamicOffsets, float
     RETURN_IF_FINISHED();
 
     auto indexSizeInBytes = (m_indexType == MTLIndexTypeUInt16 ? sizeof(uint16_t) : sizeof(uint32_t));
-    auto firstIndexOffsetInBytes = firstIndex * indexSizeInBytes;
-    auto indexBufferOffsetInBytes = m_indexBufferOffset + firstIndexOffsetInBytes;
+    auto firstIndexOffsetInBytes = checkedProduct<uint64_t>(firstIndex, indexSizeInBytes);
+    auto indexBufferOffsetInBytes = checkedSum<uint64_t>(m_indexBufferOffset, firstIndexOffsetInBytes);
+
+    if (firstIndexOffsetInBytes.hasOverflowed() || indexBufferOffsetInBytes.hasOverflowed()) {
+        makeInvalid(@"Invalid offset to drawIndexed");
+        return;
+    }
 
     if (NSString* error = errorValidatingDrawIndexed()) {
         makeInvalid(error);
         return;
     }
 
-    if (firstIndexOffsetInBytes + indexCount * indexSizeInBytes > m_indexBufferSize) {
+    auto lastIndexOffset = checkedSum<NSUInteger>(firstIndexOffsetInBytes, indexCount * indexSizeInBytes);
```

Comment:
I suppose there is a potential for overflow in `indexCount * indexSizeInBytes`, but since `indexSizeInBytes` is at most 4, you could write static_cast<uint64_t>(indexCount) * indexSizeInBytes but this is already a variable, `indexCountInBytes`. But on watchOS `indexCountInBytes` is only 32-bits, so it could overflow on watchOS on line 628.

Perhaps this isn't a concern as WebGPU does not support watchOS